users/models.py

```diff
@@ -0,0 +1,70 @@
+from django.db import models
+from django.utils import timezone
+from django.contrib.auth.models import (AbstractBaseUser,
+                                        PermissionsMixin,
+                                        BaseUserManager)
+
+
+class UserManager(BaseUserManager):
+
+    def create_user(self, email, first_name,
+                    last_name, birth_date, password,
+                    **other_fields):
+
+        if not email:
+            raise ValueError('You must provide an email address')
+
+        if not first_name:
+            raise ValueError('You must provide a first name')
+
+        if not last_name:
+            raise ValueError('You must provide a last name')
+
+        if not birth_date:
+            raise ValueError('You must provide a birth date')
+
+        email = self.normalize_email(email)
+        user = self.model(email=email, first_name=first_name,
+                          last_name=last_name, birth_date=birth_date,
+                          **other_fields)
+        user.set_password(password)
+        user.save()
+        return user
+
+    def create_superuser(self, email, first_name,
+                         last_name, birth_date, password,
+                         **other_fields):
+
+        other_fields.setdefault('is_staff', True)
+        other_fields.setdefault('is_superuser', True)
+
+        if not other_fields.get('is_staff'):
+            raise ValueError('superuser must be assigned to is_staff=True')
+
+        if not other_fields.get('is_superuser'):
+            raise ValueError('superuser must be assigned to is_superuser=True')
+
+        return self.create_user(email, first_name,
+                                last_name, birth_date, password,
+                                **other_fields)
+
+
+class User(AbstractBaseUser, PermissionsMixin):
```

Comment:
I think the `tags` field we've discussed should go here too, no?

Reply:
Isn't it an apartment feature?
I thought we will add it in the apartment profile after this PR marges.
What do you think?

Reply:
Each tag is an interest/hobby of an apartment owner or seeker, so it should go be user field in my opinion.
The tags will just be a set of strings. We'll use the sets to calculate a matching score between two users.

Reply:
I think its more of a user feature too, but both ways are supposed to work.

Reply:
IMO it should be User fields.
cuz both Seeker and Owner have them.

Reply:
Ok, ill add `tags` to the User. But because it's not a built-in field like others, I think it will be best if we merged this first and then add the `tags` field later this week, this way we can work on models that rely on the BaseUser simultaneously.

Reply:
We agreed that @nadavsu will take care of the `tags` in the base user after the merging of this PR, marking this as "resolved".